frontend/src/lib/logic/inAppPrompt/inAppPromptLogic.tsx

```diff
@@ -0,0 +1,454 @@
+                runPrompt: () => true,
+                closePrompts: () => false,
+                dismissSequence: () => false,
+            },
+        ],
+    })),
+    selectors(() => ({
+        prompts: [(s) => [s.currentSequence], (sequence: PromptSequence | null) => sequence?.prompts ?? []],
+        sequenceKey: [(s) => [s.currentSequence], (sequence: PromptSequence | null) => sequence?.key],
+    })),
+    listeners(({ actions, values, cache }) => ({
+        syncState: async ({ options }) => {
+            try {
+                const updatedState = await api.update(
+                    `api/projects/${teamLogic.values.currentTeamId}/prompts/my_prompts`,
+                    values.userState
+                )
+                if (updatedState) {
+                    if (JSON.stringify(values.userState) !== JSON.stringify(updatedState['state'])) {
+                        actions.setUserState(updatedState['state'])
+                    }
+                    if (
+                        JSON.stringify(values.sequences) !== JSON.stringify(updatedState['sequences']) ||
+                        options.forceRun
+                    ) {
+                        actions.setSequences(updatedState['sequences'])
+                    }
+                }
+            } catch (error: any) {
+                console.error(error)
+            }
+        },
+        closePrompts: () => cache.runOnClose?.(),
+        setSequences: actions.findValidSequences,
+        runSequence: ({ sequence, step = 0 }) => {
+            const prompt = sequence.prompts.find((prompt) => prompt.step === step)
+            if (prompt) {
+                actions.runPrompt(prompt)
+                if (step === sequence.prompts.length - 1) {
+                    actions.updatePromptState({
+                        completed: true,
+                    })
+                    inAppPromptEventCaptureLogic.actions.reportPromptSequenceCompleted(
+                        sequence.key,
+                        step,
+                        values.prompts.length
+                    )
+                }
+            }
+        },
+        runPrompt: ({ prompt }) => {
+            switch (prompt.type) {
+                case 'tooltip':
+                    actions.tooltip(prompt)
+                    break
+                default:
+                    break
+            }
+        },
+        tooltip: async ({ tooltip }) => {
+            const { close, show } = cancellableTooltipWithRetries(tooltip, actions.promptAction, {
+                maxSteps: values.prompts.length,
+                onClose: () => {
+                    actions.dismissSequence()
+                },
+                previous: actions.previousPrompt,
+                next: actions.nextPrompt,
+            })
+            cache.runOnClose = close
+
+            show.then(() => {
+                actions.updatePromptState({ step: values.currentStep })
+            }).catch((err) => console.error(err))
```

Comment:
Not super strongly opinionated on this but if we are in an async function perhaps we should keep things async-style and try-catch this?